### PR TITLE
Turned on horizontal fusion for small tensors

### PR DESCRIPTION
**Pull request by Chillee**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #97428
* #98659
* __->__ #98657

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @ngimel @yf225 @anijain2305 @soumith @desertfire

### CODE REVIEW

File: torch/_inductor/triton_heuristics.py [hunk 1]
```diff
@@ -602,7 +602,7 @@ def reduction(size_hints, reduction_hint=False, meta=None, filename=None):
         tiny_config = triton_config_reduction(
             size_hints, 2 * (256 // rnumel) if rnumel <= 256 else 1, min(rnumel, 2048)
         )
-        if config.max_autotune or config.max_autotune_pointwise:
+        if config.max_autotune or config.max_autotune_pointwise or True:
```
Comment (jansel) [inline torch/_inductor/triton_heuristics.py:605]: Whoops?

**Review by jansel — COMMENTED** [review]
Any perf data?

**Comment by Chillee** [comment]
@jansel The main goal of this PR is to horizontally fuse the batch-norm increments.

image

It's not that much of a gain in my testing, takes mobilenet_v2 from 1.355 to 1.360. Partially just putting up this diff so I can see how much of a perf gain it has.

**Review by ngimel — APPROVED** [review]
lint failure is real, and distributed tests check expected code which changed, but that's an easy fix.

**Review by jansel — APPROVED** [review]

**Pull request closed without merging**